src/_guides/testing.md

```diff
@@ -0,0 +1,157 @@
+
+* _Integration_ and _end-to-end_ tests verify the behavior of
+  an entire app, or a large chunk of an app. An integration test
+  generally runs on a real device or OS simulator (for mobile),
+  or on a browser (for the web), and consists of two pieces:
+  the app itself, and the test app that puts
+  it through its paces. An integration test often measures performance,
+  so the test app generally runs on a different device or OS.
+
+## Generally useful libraries
+
+How you write your tests partly depends on whether you are using Flutter,
+Dart VM, AngularDart, or other web solutions, but the following packages
+are useful across Dart platforms:
+
+* [package:test](https://pub.dartlang.org/packages/test)<br>
+  Provides a standard way of writing tests in Dart. You can use the test
+  package to:
+    * Write single tests, or groups of tests.
+    * Use the `@TestOn` annotation to restrict tests to run on
+      specific environments.
+    * Write asynchronous tests just as you would for non-asynchronous
+      tests.
+    * Tag tests using the `@Tag` annotation. For example, define a tag to
+      create a custom configuration for some tests, or to identify some tests
+      as needing more time to complete.
+    * Create a `dart_test.yaml` file to configure tagged tests across
+      multiple files or an entire package.
+
+
+* [package:mockito](https://pub.dartlang.org/packages/mockito)<br>
+  Provides a way to create
+  [mock objects,](https://en.wikipedia.org/wiki/Mock_object)
+  easily configured for use in fixed scenarios, and to verify
+  that the system under test interacts with the mock object in
+  expected ways.
+  For an example that uses both package:test and package:mockito,
+  see the [International Space Station API library and its unit
+  tests](https://github.com/dart-lang/mockito/tree/master/test/example/iss)
+  in the [mockito package](https://github.com/dart-lang/mockito).
+
+## Flutter testing
+
+Use the following resources to learn more about testing Flutter apps:
+
+* [Testing Flutter Apps](https://flutter.io/testing/)<br>
+  How to perform unit, widget, or integration tests on a Flutter app.
+* [flutter_test](https://docs.flutter.io/flutter/flutter_test/flutter_test-library.html)<br>
+  A testing library for Flutter built on top of package:test.
+* [flutter_driver](https://docs.flutter.io/flutter/flutter_driver/flutter_driver-library.html)<br>
+  A testing library for testing Flutter applications on real devices and
+  emulators (in a separate process).
+* [flutter/examples/flutter_gallery](https://github.com/flutter/flutter/tree/master/examples/flutter_gallery)<br>
+  Tests for the Flutter gallery example.
+* [flutter/dev/manual_tests](https://github.com/flutter/flutter/tree/master/dev/manual_tests)<br>
+  Many examples of tests in the Flutter SDK.
+
+## Web testing
+
+Use the following resources to learn more about testing Dart web
+applications:
+
+* [Testing](https://webdev.dartlang.org/angular/guide/testing)<br>
```

Comment:
Question for @kwalrath - are we going to make https://webdev.dartlang.org/angular/guide/testing the "Component Testing" page for now (at least until you get a chance to write the other testing page)?